awslabs/service-workbench-on-aws
# feat: Merge Egress, AppStream and Account Update features

Issue #, if available: GALI-1177

Description of changes: Merge branch `feat-secure-workspace-egress` into develop. This PR mainly merges three independent features:

1. Egress
2. Secured Workspaces (AppStream)
3. Account update wizard

Testing done:

* AppStream flag disabled GALI-1177
  * Mount internal studies and BYOB studies and test read/write functionality in Windows, SageMaker and Linux
  * Spin up Windows, SageMaker, Linux and RStudio and test for connectivity
  * Verify start/stop
  * Verify the above workspaces from old templates, and new templates. Also ensure that we test workspaces before account and after account updates

* Testing AppStream feature GALI-1171
  * Verify account update
  * Verify image builder process
  * Mount internal studies and BYOB studies and test read/write functionality in Windows, SageMaker and Linux
  * Spin up Windows, SageMaker, Linux and RStudio and test for connectivity
  * Verify auto-stop

* Testing Egress store feature GALI-1172
  * Merge AppStream feature branch into develop in your local and deploy
  * Verify write functionality of egress store from Windows, SageMaker and Linux
  * Verify cleanup of roles
  * Verify notifications are sent correctly (submit egress store functionality)
  * Verify that UI correctly reflects the egress store contents
  * Egress store cannot be disabled once enabled

Checklist:



- [x] If new dependencies have been added, have they been pinned to specific versions?
- [x] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [x] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?



AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
